include/clap/ext/draft/midi-info.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include "../../plugin.h"
+
+// This extension lets a plugin expose it's MIDI implementation.
+// The primary goal is to allow a host to create a MIDI-CI Property Exchange "ChCtrlList" resource for the plugin,
+// so keyboard controllers can assign their knobs to suitable controllers.
+
+//TODO : clap_midiinfo could contain more (optional) info (min/max, stepcount, paramPath etc.)
+//TODO : for completeness this should work per-input port. Is that worth it?
+
+static CLAP_CONSTEXPR const char CLAP_EXT_MIDIINFO[] = "clap.midi-info.draft/0";
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum {
+   //MIDI 2.0 message status
+   CLAP_MIDIINFO_STATUS_REGISTERED_PER_NOTE_CONTROLLER = 0x00,
+   CLAP_MIDIINFO_STATUS_ASSIGNABLE_PER_NOTE_CONTROLLER = 0x10,
+   CLAP_MIDIINFO_STATUS_RPN = 0x20,
+   CLAP_MIDIINFO_STATUS_NRPN = 0x30,
+   CLAP_MIDIINFO_STATUS_PER_NOTE_PITCHBEND = 0x60,
+   CLAP_MIDIINFO_STATUS_POLY_AFTERTOUCH = 0xA0,
+   CLAP_MIDIINFO_STATUS_CONTROLLER = 0xB0,
+   CLAP_MIDIINFO_STATUS_CHANNEL_AFTERTOUCH = 0xD0,
+   CLAP_MIDIINFO_STATUS_PITCHBEND = 0xE0,
+
+   //CLAP note expressions.
+   //A host may map MIDI messages to note expressions, so it needs to know which note expressions are supported.
+   CLAP_MIDIINFO_STATUS_NOTE_EXPRESSION_VOLUME = 0x100,
+   CLAP_MIDIINFO_STATUS_NOTE_EXPRESSION_PAN = 0x101,
+   CLAP_MIDIINFO_STATUS_NOTE_EXPRESSION_TUNING = 0x102,
+   CLAP_MIDIINFO_STATUS_NOTE_EXPRESSION_VIBRATO = 0x103,
+   CLAP_MIDIINFO_STATUS_NOTE_EXPRESSION_EXPRESSION = 0x104,
+   CLAP_MIDIINFO_STATUS_NOTE_EXPRESSION_BRIGHTNESS = 0x105,
+   CLAP_MIDIINFO_STATUS_NOTE_EXPRESSION_PRESSURE = 0x106
+};
+
+/* This describes a MIDI control/message */
+typedef struct clap_midiinfo {
+   //CLAP_MIDIINFO_STATUS, describes what kind of control this struct is about.
+   status: uint32_t;        

```

Comment:
This syntax is backwards from the syntax we use everywhere else, namely `uint32_t status`. I didn't even know the postfix type worked with C! but I think we should be consistent.

Reply:
Oops, that's Delphi code :-)
Yet github seems to be able to compile it. Odd.

Reply:
well this .h file isn't included anywhere is it? so I'm not sure the compiler even ran it.

Reply:
Yeah unless you are included <https://github.com/free-audio/clap/blob/41964fa3f44fdbf6feb20df528a9cdf2cc3b05ba/include/clap/clap.h#L64> here you won't get compiled by CI.